airbyte-ci/connectors/live-tests/src/live_tests/commons/connection_objects_retrieval.py

```diff
@@ -0,0 +1,125 @@
+# Copyright (c) 2024 Airbyte, Inc., all rights reserved.
+
+import json
+import logging
+from pathlib import Path
+from typing import Dict, Optional, Set
+
+from connection_retriever import ConnectionObject, retrieve_objects  # type: ignore
+
+from .models import Command, ConfiguredAirbyteCatalog, ConnectionObjects, SecretDict
+
+LOGGER = logging.getLogger(__name__)
+
+
+def parse_config(config: Dict | str | None) -> Optional[SecretDict]:
+    if not config:
+        return None
+    if isinstance(config, str):
+        return SecretDict(json.loads(config))
+    else:
+        return SecretDict(config)
+
+
+def parse_catalog(catalog: Dict | str | None) -> Optional[ConfiguredAirbyteCatalog]:
+    if not catalog:
+        return None
+    if isinstance(catalog, str):
+        return ConfiguredAirbyteCatalog.parse_obj(json.loads(catalog))
+    else:
+        return ConfiguredAirbyteCatalog.parse_obj(catalog)
+
+
+def parse_state(state: Dict | str | None) -> Optional[Dict]:
+    if not state:
+        return None
+    if isinstance(state, str):
+        return json.loads(state)
+    else:
+        return state
+
+
+CONNECTION_OBJECT_TYPE_TO_PARSER = {
+    ConnectionObject.SOURCE_CONFIG: parse_config,
+    ConnectionObject.CONFIGURED_CATALOG: parse_catalog,
+    ConnectionObject.STATE: parse_state,
+}
```

Comment:
I don't see where we use this or `COMMAND_TO_REQUIRED_OBJECT_TYPES` (though they seem helpful!)

Reply:
It's used here:
https://github.com/airbytehq/airbyte/blob/f3f81124ee8a8cfa7455c2d72847b6950a4028c3/airbyte-ci/connectors/live-tests/src/live_tests/debug/cli.py#L91